tasks/src/main/java/com/example/Quickstart.java

```diff
@@ -0,0 +1,173 @@
+      .hasArg()
+      .argName("project-id")
+      .type(String.class)
+      .build();
+
+  private static Option QUEUE_OPTION = Option.builder("queue")
+      .required()
+      .longOpt("queue")
+      .desc("The Cloud Tasks queue.")
+      .hasArg()
+      .argName("queue")
+      .type(String.class)
+      .build();
+
+  private static Option LOCATION_OPTION = Option.builder("location")
+      .required()
+      .longOpt("location")
+      .desc("The region in which your queue is running.")
+      .hasArg()
+      .argName("location")
+      .type(String.class)
+      .build();
+
+  public static void main(String... args) throws Exception {
+    Options options = new Options();
+    options.addOption(PROJECT_ID_OPTION);
+    options.addOption(QUEUE_OPTION);
+    options.addOption(LOCATION_OPTION);
+
+    if (args.length == 0) {
+      printUsage(options);
+      return;
+    }
+
+    CommandLineParser parser = new DefaultParser();
+    CommandLine params;
+    try {
+      params = parser.parse(options, args);
+    } catch (ParseException e) {
+      System.err.println("Invalid command line: " + e.getMessage());
+      printUsage(options);
+      return;
+    }
+
+    String projectId;
+    if (params.hasOption("project-id")) {
+      projectId = params.getOptionValue("project-id");
+    } else {
+      projectId = System.getenv(GGOGLE_CLOUD_PROJECT_KEY);
+    }
+    if (Strings.isNullOrEmpty(projectId)) {
+      printUsage(options);
+      return;
+    }
+
+    String queue = params.getOptionValue(QUEUE_OPTION.getOpt());
+    String location = params.getOptionValue(LOCATION_OPTION.getOpt());
+
+    switch (args[0]) {
+      default:
+        printUsage(options);
+        break;
+      case "create-task":
+        createTask(projectId, queue, location);
+        break;
+      case "lease-and-ack-task":
+        pullAndAckTask(projectId, queue, location);
+        break;
+    }
+  }
+
+  // [START cloud_tasks_create_task]
+  private static void createTask(String projectId, String queueName, String location)
+      throws IOException {
+    try (CloudTasksClient client = CloudTasksClient.create()) {
+      Task.Builder taskBuilder = Task
+          .newBuilder()
+          .setPullMessage(PullMessage.newBuilder().setPayload(
+              ByteString.copyFrom("a message for recipient", Charset.defaultCharset())));
+
+      Task newTask = client.createTask(CreateTaskRequest
+          .newBuilder()
+          .setParent(QueueName.of(projectId, location, queueName).toString())
+          .setTask(taskBuilder)
+          .build());
+      System.out.println("Task created: " + newTask.getName());
+    }
+  }
+  // [END cloud_tasks_create_task]
+
+  // [START cloud_tasks_lease_and_acknowledge_task]
+  private static void pullAndAckTask(String projectId, String queueName, String location) {
+    try (CloudTasksClient client = CloudTasksClient.create()) {
+      LeaseTasksRequest.Builder reqBuilder = LeaseTasksRequest.newBuilder()
```

Comment:
Any reason you are passing a builder instead of just the request?

Reply:
I'm not, on line 148 it passes `reqBuilder.build()`

Reply:
Yup that's what I meant - typically the builder pattern is a replacement for `new`, so it seems weird to trigger the build there.